datalad/support/annexrepo.py

```diff
@@ -335,3 +336,103 @@ def annex_add_to_git(self, files):
             self.cmd_call_wrapper.run(cmd_list, cwd=self.path)
         else:
             self.git_add(files)
+
+    def annex_initremote(self, name, options):
+        """Creates a new special remote
+
+        Parameters:
+        -----------
+        name: str
+            name of the special remote
```

Comment:
1. Direct mode or not, an annex is still a git repo. And you can use it as
such from command line as well as via python-interface.

2. Well, this would require an additional interface layer either way, I
think. So just a renaming.

2015-04-15 15:27 GMT+02:00 Yaroslav Halchenko <notifications@github.com>:

> In datalad/support/annexrepo.py
> <https://github.com/yarikoptic/datalad/pull/2#discussion_r28418462>:
>
> > @@ -335,3 +336,103 @@ def annex_add_to_git(self, files):
> >              self.cmd_call_wrapper.run(cmd_list, cwd=self.path)
> >          else:
> >              self.git_add(files)
> > +
> > +    def annex_initremote(self, name, options):
> > +        """Creates a new special remote
> > +
> > +        Parameters:
> > +        -----------
> > +        name: str
> > +            name of the special remote
>
> may be may be... I am still undecided though either AnnexRepo should be a
> GitRepo at all given all the direct-mode-ness ;)
>
> In the long shot I also see Dataset not being an AnnexRepo either -- may
> be we would become fancy to support various backends...? ;)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/yarikoptic/datalad/pull/2/files#r28418462>.
>